legacy/builder/wipeout_build_path_if_build_options_changed.go

```diff
@@ -40,9 +42,16 @@ func (s *WipeoutBuildPathIfBuildOptionsChanged) Run(ctx *types.Context) error {
 	previousBuildOptionsJson := ctx.BuildOptionsJsonPrevious
 
 	var opts *properties.Map
+	if err := json.Unmarshal([]byte(buildOptionsJson), &opts); err != nil || opts == nil {
+		ctx.GetLogger().Println(constants.LOG_LEVEL_DEBUG, constants.MSG_BUILD_OPTIONS_INVALID, constants.BUILD_OPTIONS_FILE)
```

Comment:
I think this message is confusing:
>build.options.json invalid

Because it's not the build.options.json file that's invalid, but rather the internal build options data.

Since this isn't something that should occur under any normal circumstances, I would lean toward just doing a `panic(err)`. Probably @silvanocerza would be able to tell us what is the preferred approach though.

Reply:
I think it's fine like this, it seems to me that `ctx.BuildOptionsJson` is populated with data from the `build.options.json` file only.

Reply:
The `build.options.json` file is created from `ctx.BuildOptionsJson`:
https://github.com/arduino/arduino-cli/blob/079bb6c66505193d61366f355003b28ce177904b/legacy/builder/store_build_options_map.go#L26
It's `ctx.BuildOptionsJsonPrevious` that is populated with data from the `build.options.json` file:
https://github.com/arduino/arduino-cli/blob/079bb6c66505193d61366f355003b28ce177904b/legacy/builder/load_previous_build_options.go#L38

Reply:
Should I leave it as it currently is,
or change the message,
or replace it by `panic(constants.BUILD_OPTIONS_FILE + " is invalid")` ?

Reply:
I'd like to get @silvanocerza's response to my previous reply just to make sure everyone is on the same page here. After that, I'd recommend going with whatever Silvano think is best.

Reply:
@silvanocerza @per1234 I just proceeded with your suggestion.